model/windowInformation.cc

```diff
@@ -67,44 +71,68 @@ namespace minsky
   }
 #endif
 
-  unsigned long WindowInformation::getChildWindowId() {
-    return childWindowId;
-  }
-
-  Display *WindowInformation::getDisplay() {
+  Display *WindowInformation::getDisplay()
+  {
     return display;
   }
 
-  WindowInformation::~WindowInformation() {
-    childSurface.reset();
+  WindowInformation::~WindowInformation()
+  {
+    windowSurface.reset();
+    bufferSurface.reset();
     XDestroyWindow(display, childWindowId);
   }
 
-  ecolab::cairo::SurfacePtr WindowInformation::getSurface() {
-    return childSurface;
+  ecolab::cairo::SurfacePtr WindowInformation::getBufferSurface()
+  {
+    return bufferSurface;
   }
 
-  void WindowInformation::createSurface(){
+  void WindowInformation::copyBufferToMain()
+  {
+    cairo_set_source_surface(windowSurface->cairo(), bufferSurface->surface(), 0, 0);
+    cairo_paint(windowSurface->cairo());
```

Comment:
I am sceptical this will be performant enough. We'll see.

Reply:
Please do let me know once you have tested. It seemed to work good enough when I and Ravi tested. Maybe there is an even faster way but this seemed simple and worked.